Modules/_testcapi/complex.c

```diff
@@ -1,7 +1,95 @@
 #include "parts.h"
 #include "util.h"
 
+
+static PyObject *
+complex_check(PyObject *Py_UNUSED(module), PyObject *obj)
+{
+    NULLABLE(obj);
+    return PyLong_FromLong(PyComplex_Check(obj));
+}
+
+static PyObject *
+complex_checkexact(PyObject *Py_UNUSED(module), PyObject *obj)
+{
+    NULLABLE(obj);
+    return PyLong_FromLong(PyComplex_CheckExact(obj));
+}
+
+static PyObject *
+complex_fromccomplex(PyObject *Py_UNUSED(module), PyObject *obj)
+{
+    Py_complex complex = ((PyComplexObject*)obj)->cval;
```

Comment:
Do not access PyComplexObject internals directly. Either use `PyComplex_AsCComplex()` or `PyArg_Parse("C")`, or create `Py_complex` from a pair of doubles, like in `complex_fromdoubles()`.

If you make it creating from a pair of doubles, you can also make `complex_asccomplex()` returning a pair of doubles, so every wrapper will use only one PyComplex C API function.

Reply:
I think, ``PyArg_Parse("D")`` (give ``Py_complex``) is the best way.